engine/Cuboid.js

```diff
@@ -0,0 +1,78 @@
+class Cuboid {
+  constructor(width, height, depth, options = {
+    position: [0, 0, 0], rotation: [0, 0, 0]
+  }) {
+    this.width = width;
+    this.height = height;
+    this.depth = depth;
+    this.position = options.position;
+    // this.rotation = options.rotation;
```

Comment:
the rotation is not part of a mesh (or at least not of a generated one), it's expressed in the model itself and then converted into a matrix. Finally, the vertex shader takes care of rotating.

Reply:
Why should you not be able to have an already rotated mesh? A mesh is just triangles.

Reply:
The main problem of an already rotated/translated mesh is that you're offsetting the object from its origin. Let's say, you translated the object [0, 0, -1]. If you now rotate it, it will still rotate around [0, 0, 0] and not around the center of the object. This leads to unexpected transformation results. You could still scale the object though :).
Furthermore, if you have unique geometry for each object, you have to first generate it every time for each model which increases the memory overhead and prevents you from reusing similar geometry during rendering.